hevc_codec_registration.src.html

```diff
@@ -0,0 +1,212 @@
+
+VideoDecoderConfig description {#videodecoderconfig-description}
+================================================================
+
+If the {{VideoDecoderConfig.description}} is present, it is assumed to be an
+`HEVCDecoderConfigurationRecord`, as defined by [[iso14496-15]], section
+8.3.3.1, and the bitstream is assumed to be in {{HevcBitstreamFormat/hevc}}
+format.
+
+NOTE: This format is commonly used in .MP4 files, where the player generally
+    has random access to the media data.
+
+If the {{VideoDecoderConfig.description}} is not present, the bitstream is
+assumed to be in {{HevcBitstreamFormat/annexb}} format.
+
+NOTE: "annexb" format is described in greater detail by [[ITU-T-REC-H.265]],
+    Annex B. This format is commonly used in live-streaming applications, where
+    including the VPS, SPS, and PPS data periodically allows users to easily
+    start from the middle of the stream.
+
+EncodedVideoChunk type {#encodedvideochunk-type}
+================================================
+
+If an {{EncodedVideoChunk}}'s [=EncodedVideoChunk/[[type]]=] is
+{{EncodedVideoChunkType/key}}, and the bitstream is in
+{{HevcBitstreamFormat/hevc}} format, then the {{EncodedVideoChunk}} is expected
+to contain a base layer primary coded picture that is an instantaneous decoding
+refresh (IDR), clean random access (CRA), or broken link access (BLA) picture.
+
+NOTE: If the bitstream is in {{HevcBitstreamFormat/hevc}} format, parameter sets
+    necessary for decoding are included in {{VideoDecoderConfig.description}}.
+
+If an {{EncodedVideoChunk}}'s [=EncodedVideoChunk/[[type]]=] is
+{{EncodedVideoChunkType/key}}, and the bitstream is in
+{{HevcBitstreamFormat/annexb}} format, then the {{EncodedVideoChunk}} is
+expected to contain both a base layer coded picture that is an instantaneous
+decoding refresh (IDR), clean random access (CRA), or broken link access (BLA)
+picture, and all parameter sets necessary to decode all video data
+NAL units in the {{EncodedVideoChunk}}.
+
+VideoEncoderConfig extensions {#videoencoderconfig-extensions}
+==============================================================
+
+<pre class='idl'>
+<xmp>
+partial dictionary VideoEncoderConfig {
+  HevcEncoderConfig hevc;
+};
+</xmp>
+</pre>
+
+<dl>
+  <dt><dfn dict-member for=VideoEncoderConfig>hevc</dfn></dt>
+  <dd>
+    Contains codec specific configuration options for the HEVC (H.265) codec.
+  </dd>
+</dl>
+
+HevcEncoderConfig {#hevc-encoder-config}
+--------------------------------------
+<pre class='idl'>
+<xmp>
+dictionary HevcEncoderConfig {
+  HevcBitstreamFormat format = "hevc";
+};
+</xmp>
+</pre>
+
+<dl>
+  <dt><dfn dict-member for=HevcEncoderConfig>format</dfn></dt>
+  <dd>
+    Configures the format of output {{EncodedVideoChunk}}s. See
+    {{HevcBitstreamFormat}}.
+  </dd>
+</dl>
+
+HevcBitstreamFormat {#hevc-bitstream-format}
+------------------------------------------
+<pre class='idl'>
+<xmp>
+enum HevcBitstreamFormat {
+  "annexb",
+  "hevc",
```

Comment:
I'm not sure if there is a common name for this. For H.264 I think it was commonly called "AVC format" leading to the format string "avc", but perhaps this one should be called "bmff" which would be more precise.

Reply:
"hevc" might be more familiar.

Reply:
+1 for hevc, seems less confusing.